packages/amplify_datastore_plugin_interface/lib/src/types/models/model_association.dart

```diff
@@ -14,17 +14,19 @@
  */
 import 'dart:convert';
 
+enum ModelAssociationEnum { hasMany, hasOne, belongsTo }
+
 class ModelAssociation {
-  final String name;
-  final String targetName;
-  final String associatedName;
-  final String associatedType;
+  ModelAssociationEnum associationType;
+  String targetName; // opt (used in belongsTo)
+  String associatedName; // opt (used in hasMany/hasOne)
+  String associatedType; // opt (used in hasMany/hasOne)
 
-  const ModelAssociation({
-    this.name,
-    this.targetName,
-    this.associatedName,
-    this.associatedType,
+  ModelAssociation({
+    this.associationType,
+    this.targetName = "",
+    this.associatedName = "",
+    this.associatedType = "",

```

Comment:
empty strings feel like the wrong default. If these are not specified, they should probably be nullable.

Reply:
The strings are empty because null is not serializable on the method channel.  

When serializing data between Dart and Android, I decided to include all keys.  If a key is "null" I just send an empty string instead.  This is to simplify the code.  

What are your thoughts on that general approach @jamesonwilliams ?

Reply:
What's the complications in null checks in de/serialization?

Reply:
I understand why you've done it this way, now, thanks. I think that works. The other approach you could consider would be to not send keys associated to null values. Is there any downside to that approach?

That is, if you have:
```kotlin
val map = mapOf("key1" to null, "key2" to "Hamburger", "key3" to 5)
```
You can either do:
```kotlin
dispatchToMethodChannel(
    map.filterValues { it != null }
)
```
or:
```kotlin
dispatchToMethodChannel(
    map.mapValues { if (it == null) "" else it }
)
```

This probably isn't our most pressing issue right now though. Feel free to ignore in favor of higher value work items. :-D

Reply:
Thanks for that suggestion @jamesonwilliams.  I don't think that Dart has such elegant syntax for map comprehensions like Kotlin, but I found a similar way in Dart here: https://stackoverflow.com/questions/21130889/how-can-i-filter-a-map-access-the-entries

I agree it's not our most pressing issue and I don't see a strong pull to either solution beyond possibly adding more burden on what needs to be passed back and forth on the method channel.  

The benefit I see is simpler code on the Dart side that doesn't need to worry about checking for null values being sent or stored.  Also, I remember speaking with an Android developer and they told me that passing null or empty values to the Kotlin builders wasn't the ideal thing to do.  Can you confirm if that's true @jamesonwilliams ? 

@Amplifiyer it's not a big complication to do those null checks, but they do require extra checks for every single field which can make our methods larger.

Reply:
Ah, true @fjnoyp, sorry, I was in Kotlin-mode, but this is clearly Dart.

I guess Dart 2.0 has a `removeWhen` function:
```dart
someMap.removeWhere((key, value) => value == null);
```